apps/explorer/lib/explorer/chain.ex

```diff
@@ -4102,7 +4151,7 @@ defmodule Explorer.Chain do
   @doc """
   The current total number of coins minted minus verifiably burned coins.
   """
-  @spec total_supply :: non_neg_integer() | nil
+  @spec total_supply :: Decimal.t() | nil
```

Comment:
Could you please clarify what is the rationale of changing spec of output type? And it looks like 0 which is one of the possible output doesn't match with Decimal type.

Reply:
Dialyzer complained because `total_supply` is defined as `Decimal.t()` in  `Explorer.ExchangeRates.Token`. I could change it to `Decimal.t() | 0 | nil`?